### PR TITLE
ref(worker): move env var processing to the top

### PR DESCRIPTION
**Pull request by technosophos**
This is a minor refactor to move an env var reference to the top.

### CODE REVIEW

**Comment by technosophos** [comment]
I'm trying to figure out how to best bubble this env var reference up to the `index.js` file. This feels pretty hacky, so if you have any clever ideas, I'd be happy to make this better.

/cc @jchanam and @adamreese

**Comment by jchanam** [comment]
I think this is a much better approach. The only thing that I don't know is: Why to have the default value defined twice?

This is what I tried to do on my first approach, but couldn't achieve it :(

**Comment by mumoshu** [comment]
@technosophos Just trying to figure out your intension -
 maybe you're going to update this line: https://github.com/Azure/brigade/pull/358/files#diff-d54f19b6bbea870b89fa594044af83d9R158

from

```
this.serviceAccount = job.serviceAccount || process.env.BRIGADE_SERVICE_ACCOUNT || defaultServiceAccount;
```

to

```
this.serviceAccount = job.serviceAccount || defaultServiceAccount;
```

on the other hand of your current change?

**Comment by technosophos** [comment]
Ugh... that was my mistake, @mumoshu. I need to fix that.

**Comment by technosophos** [comment]
Okay, refactored this based on some OTS input from @adamreese . This  new version should be a little bit more future-proof, and is based on an idea he found in `ts-node`.

File: brigade-worker/src/k8s.ts [hunk 1]
```diff
@@ -142,18 +166,18 @@ export class JobRunner implements jobs.JobRunner {
   event: BrigadeEvent;
   job: jobs.Job;
   client: kubernetes.Core_v1Api;
+  localOptions: KubernetesOptions;
```
Comment (adamreese) [inline brigade-worker/src/k8s.ts]: i would just call this `options`

Comment (technosophos) [reply brigade-worker/src/k8s.ts]: Fixed

**Pull request merged**